tracing-subscriber/src/filter/env/field.rs

```diff
@@ -36,8 +38,19 @@ pub(crate) struct MatchVisitor<'a> {
     inner: &'a SpanMatch,
 }
 
+/// Specifies how a [field] [value] is matched when applying directives to [span]s.
+///
+/// [span]: mod@tracing::span
+/// [field]: fn@tracing::Metadata::fields
+/// [value]: tracing#recording-fields
+#[derive(Debug, Clone)]
+#[repr(transparent)]
+pub struct ValueMatch {
+    inner: ValueMatchInternal,
```

Comment:
For building directives this needs to be a public type in some form, the reason why I choose a wrapper instead of  `#[non_exhaustive]` is:

- like #[non_exhaustive] we can add variants
- but we also can fully change the internal representation
- and we don't want to expose `MatchDebug` and `MatchPattern`
  - through we could give them a public interface, too
- we wan to prevent people from creating a `F64(f64::NAN)`

I still could change it to a  pub `#[non_exhaustive]` enum if desired.